src/components/libs/featurevisor/Featurevisor.instance.brs

```diff
@@ -713,18 +713,24 @@ sub refresh()
   m._statuses.refreshInProgress = true
 
   chain = createRequest("FeaturevisorRequest", { datafileUrl: m._datafileUrl })
-  chain.then(sub (datafile as Object, m as Object)
+  chain.then(function(datafile as object, m as object)
+    oldRevisionNumber = getRevision()
+    setDatafile(datafile)
+
+    return {
+      oldRevisionNumber: oldRevisionNumber,
+      datafile: datafile,
+    }
+  end function, sub (error as object, _m as object)
+    print "Featurevisor - failed to refresh datafile: ";error
+  end sub, m).then(sub (data as object)
```

Comment:
Let me just suggest a formatting change.
Also now you don't need context for the first callback, but you need it for the second.
```suggestion
  chain.then(function (datafile as Object)
    previousRevisionNumber = getRevision()
    setDatafile(datafile)

    return {
      datafile: datafile,
      previousRevisionNumber: previousRevisionNumber,
    }
  end function, sub (error as Object)
    print "Featurevisor - failed to refresh datafile: ";error
  end sub).then(sub (data as Object, m as Object)
```